src/services/database.ts

```diff
@@ -127,3 +129,34 @@ export const postTechRecord = async (request: TechrecordGet): Promise <Techrecor
     throw new Error('database client failed getting data');
   }
 };
+export const archiveOldCreateCurrentRecord = async (recordToArchive: TechrecordGet, recordToCreate: TechrecordGet): Promise<string | Error> => {
+  logger.info('Preparing Transact Items');
+
+  const transactParams: TransactWriteItemsInput = {
+    TransactItems: [
+      {
+        Put: {
+          TableName: tableName,
+          Item: marshall(recordToArchive),
+          ConditionExpression: 'attribute_exists(systemNumber) AND attribute_exists(createdTimestamp)',
+        },
+      },
+      {
+        Put: {
+          TableName: tableName,
+          Item: marshall(recordToCreate),
+        },
+      },
+    ],
+  };
+
+  try {
+    await ddbClient.send(new TransactWriteItemsCommand(transactParams));
+
+    return 'records updated';
```

Comment:
Does this just update one record? Would it be better to return the entire updated record?

Reply:
So currently V1 just returns a success message. which is different wording to this one. I'll consult and see what the feeling is for changing this to return the amended records.